File: Standard/C/Python3/2.py
# Create Time: 2020.11.26
# By: LovelyAnQi

x, mul, k, prime = int(input()), 1, 0, 1

while mul < x:
    i = int()
    while i * i <= prime:
        prime += 1
        i = 2
        while i * i < prime:
            if prime % i == 0:
                break
            i += 1
    mul *= prime
    k += 1

print(k)

File: Standard/C/Cpp/2.cpp
/**
 * Create Time: 2020.11.25
 * By: LovelyAnQi
 */
#include <iostream>
using namespace std;

#define ll long long

ll currentPrime = 1;

ll nextPrime() {

	for (++currentPrime;; ++currentPrime) {
		ll i;
		for (i = 2; i * i < currentPrime; ++i) {
			if (currentPrime % i == 0) {
				break;
			}
		}
		if (i * i > currentPrime) {
			break;
		}
	}
	return currentPrime;
}

int main()
{
	ll x, mul = 1, k = 0;
	cin >> x;

	while (mul < x) {
		mul *= nextPrime();
		if (++k == 16) {
			break;
		}
	}

	cout << k << endl;

	return 0;
}
